snapcraft/internal/lifecycle.py

```diff
@@ -454,27 +454,30 @@ def _verify_dependents_will_be_cleaned(part_name, clean_part_names, step,
                                        config):
     # Get the name of the parts that depend upon this one
     dependents = config.parts.get_dependents(part_name)
+    additional_dependents = []
 
     # Verify that they're either already clean, or that they will be cleaned.
     if not dependents.issubset(clean_part_names):
         for part in config.all_parts:
             if part.name in dependents and not part.is_clean(step):
                 humanized_parts = formatting_utils.humanize_list(
                     dependents, 'and')
+                additional_dependents.append(part_name)
 
-                raise errors.SnapcraftEnvironmentError(
-                    'Requested clean of {!r} but {} depend{} upon it. Please '
-                    "add each to the clean command if that's what you "
-                    'intended.'.format(part_name, humanized_parts,
-                                       's' if len(dependents) == 1 else ''))
+                logger.warning(
+                    'Requested clean of {!r} which require also cleaning '
+                    'the part{} {}'.format(part_name,
+                                           '' if len(dependents) == 1 else 's',
+                                           humanized_parts))
```

Comment:
So the exception becomes a warning? Does it mean it'll go ahead and clean parts or does it still skip?